.drone.yml

```diff
@@ -0,0 +1,1100 @@
+
+volumes:
+  - name: dockersock
+    temp: {}
+
+---
+kind: pipeline
+type: kubernetes
+name: docker-cron
+
+trigger:
+  cron:
+    - docker-cron
+
+workspace:
+  path: /tmp
+
+clone:
+  disable: true
+
+steps:
+  - name: Set up variables and Dockerfile
+    image: alpine
+    environment:
+      # increment these variables when a new major/minor version is released to bump the automatic builds
+      CURRENT_VERSION_ROOT: 4.2
+      PREVIOUS_VERSION_ONE_ROOT: 4.1
+      PREVIOUS_VERSION_TWO_ROOT: 4.0
+      DOCKERFILE_BRANCH: gus/add-drone-pipeline
+    commands:
+      - apk --update --no-cache add curl git
+      - mkdir -p /tmp/build && cd /tmp/build
+      # CURRENT_VERSION
+      - echo $(git ls-remote --tags https://github.com/gravitational/teleport | cut -d'/' -f3 | grep $CURRENT_VERSION_ROOT | grep -Ev '(alpha|beta|dev|rc)' | sort -rV | head -n1) > /tmp/build/CURRENT_VERSION_TAG.txt
+      - echo "$(cat /tmp/build/CURRENT_VERSION_TAG.txt | cut -d. -f1-2 | cut -dv -f2)" > /tmp/build/CURRENT_VERSION_TAG_GENERIC.txt
+      # PREVIOUS_VERSION_ONE
+      - echo $(git ls-remote --tags https://github.com/gravitational/teleport | cut -d'/' -f3 | grep $PREVIOUS_VERSION_ONE_ROOT | grep -Ev '(alpha|beta|dev|rc)' | sort -rV | head -n1) > /tmp/build/PREVIOUS_VERSION_ONE_TAG.txt
+      - echo "$(cat /tmp/build/PREVIOUS_VERSION_ONE_TAG.txt | cut -d. -f1-2 | cut -dv -f2)" > /tmp/build/PREVIOUS_VERSION_ONE_TAG_GENERIC.txt
+      # PREVIOUS_VERSION_TWO
+      - echo $(git ls-remote --tags https://github.com/gravitational/teleport | cut -d'/' -f3 | grep $PREVIOUS_VERSION_TWO_ROOT | grep -Ev '(alpha|beta|dev|rc)' | sort -rV | head -n1) > /tmp/build/PREVIOUS_VERSION_TWO_TAG.txt
+      - echo "$(cat /tmp/build/PREVIOUS_VERSION_TWO_TAG.txt | cut -d. -f1-2 | cut -dv -f2)" > /tmp/build/PREVIOUS_VERSION_TWO_TAG_GENERIC.txt
+      - for FILE in /tmp/build/*.txt; do echo $FILE; cat $FILE; done
+      # get Dockerfile
+      - curl -Ls -o /tmp/build/Dockerfile-cron https://raw.githubusercontent.com/gravitational/teleport/$DOCKERFILE_BRANCH/build.assets/Dockerfile-cron
+
+  - name: Build and push Teleport containers (CURRENT_VERSION)
+    image: docker:dind
+    environment:
+      OS: linux
+      ARCH: amd64
+      REPO: quay.io
+    settings:
+      username:
+        from_secret: PRODUCTION_QUAYIO_DOCKER_USERNAME
+      password:
+        from_secret: PRODUCTION_QUAYIO_DOCKER_PASSWORD
+    volumes:
+      - name: dockersock
+        path: /var/run
+    commands:
+      - export VERSION_TAG=$(cat /tmp/build/CURRENT_VERSION_TAG.txt)
+      - export OSS_IMAGE_NAME="$REPO/gravitational/teleport:$(cat /tmp/build/CURRENT_VERSION_TAG_GENERIC.txt)"
+      - export ENT_IMAGE_NAME="$REPO/gravitational/teleport-ent:$(cat /tmp/build/CURRENT_VERSION_TAG_GENERIC.txt)"
+      - export ENT_FIPS_IMAGE_NAME="$REPO/gravitational/teleport-ent:$(cat /tmp/build/CURRENT_VERSION_TAG_GENERIC.txt)-fips"
+      - docker login -u="$PLUGIN_USERNAME" -p="$PLUGIN_PASSWORD" $REPO
```

Comment:
should this be `PRODUCTION_QUAYIO_DOCKER_USERNAME` and `PRODUCTION_QUAYIO_DOCKER_PASSWORD`?

Reply:
Which part?

The use of the production credentials here is correct as we’re pushing to the `quay.io/gravitational/teleport` Docker repo. This is done by the cron job within Drone so again, credentials can’t be accessed by a PR.

The use of `PLUGIN_USERNAME` and `PLUGIN_PASSWORD` is the way Drone expands variables passed in using the `settings:` element (basically prepend with ‘PLUGIN_`, convert the names to uppercase and spaces to underscores)

Reply:
Ah, I was confused because the stages above pass username/pass via `environment`, not `settings`.
It'd be great to keep this consistent.

Reply:
In general `settings:` is intended to be used with Drone plugins (which are coded to know how to process the `PLUGIN_VARIABLE` format) whereas `environment:` is for arbitrary images which you're passing data to. They are fairly interchangable because you can just look at `${PLUGIN_VARIABLE}` rather than `${VARIABLE}` if you want.

I've changed `drone-cache` to use the `settings` format instead of `environment`.